webrtc-c/canary/src/Config.cpp

```diff
@@ -201,4 +221,114 @@ STATUS Config::init(INT32 argc, PCHAR argv[])
+    pResult->initialized = TRUE;
+}
+
+STATUS Config::initWithJSON(PCHAR filePath)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    UINT64 size;
+    jsmn_parser parser;
+    int r;
+    BYTE raw[MAX_CONFIG_JSON_FILE_SIZE];
+    Config::Value<UINT64> logLevel64;
+
+    CHK_STATUS(readFile(filePath, TRUE, NULL, &size));
+    CHK_ERR(size < 1024, STATUS_INVALID_ARG_LEN, "File size too big. Max allowed is 1024 bytes");
+    CHK_STATUS(readFile(filePath, TRUE, raw, &size));
+
+    jsmn_init(&parser);
+    jsmntok_t tokens[MAX_CONFIG_JSON_TOKENS];
+
+    r = jsmn_parse(&parser, (PCHAR) raw, size, tokens, MAX_CONFIG_JSON_TOKENS);
+    for (UINT32 i = 0; i < (UINT32) r; i++) {
+        if (compareJsonString((PCHAR) raw, &tokens[i], JSMN_STRING, (PCHAR) CANARY_CHANNEL_NAME_ENV_VAR)) {
+            jsonString(raw, tokens[++i], &channelName);
+        } else if (compareJsonString((PCHAR) raw, &tokens[i], JSMN_STRING, (PCHAR) CANARY_CLIENT_ID_ENV_VAR)) {
+            jsonString(raw, tokens[++i], &clientId);
+        } else if (compareJsonString((PCHAR) raw, &tokens[i], JSMN_STRING, (PCHAR) CANARY_TRICKLE_ICE_ENV_VAR)) {
+            jsonBool(raw, tokens[++i], &trickleIce);
+        } else if (compareJsonString((PCHAR) raw, &tokens[i], JSMN_STRING, (PCHAR) CANARY_IS_MASTER_ENV_VAR)) {
+            jsonBool(raw, tokens[++i], &isMaster);
+        } else if (compareJsonString((PCHAR) raw, &tokens[i], JSMN_STRING, (PCHAR) CANARY_USE_TURN_ENV_VAR)) {
+            jsonBool(raw, tokens[++i], &useTurn);
+        } else if (compareJsonString((PCHAR) raw, &tokens[i], JSMN_STRING, (PCHAR) CANARY_LOG_GROUP_NAME_ENV_VAR)) {
+            jsonString(raw, tokens[++i], &logGroupName);
+        } else if (compareJsonString((PCHAR) raw, &tokens[i], JSMN_STRING, (PCHAR) CANARY_LOG_STREAM_NAME_ENV_VAR)) {
+            jsonString(raw, tokens[++i], &logStreamName);
+        } else if (compareJsonString((PCHAR) raw, &tokens[i], JSMN_STRING, (PCHAR) CANARY_DURATION_IN_SECONDS_ENV_VAR)) {
+            jsonUint64(raw, tokens[++i], &duration);
+            duration.value *= HUNDREDS_OF_NANOS_IN_A_SECOND;
+        } else if (compareJsonString((PCHAR) raw, &tokens[i], JSMN_STRING, (PCHAR) CANARY_ITERATION_IN_SECONDS_ENV_VAR)) {
+            jsonUint64(raw, tokens[++i], &iterationDuration);
+            iterationDuration.value *= HUNDREDS_OF_NANOS_IN_A_SECOND;
+        } else if (compareJsonString((PCHAR) raw, &tokens[i], JSMN_STRING, (PCHAR) CANARY_FORCE_TURN_ENV_VAR)) {
+            jsonBool(raw, tokens[++i], &forceTurn);
+        } else if (compareJsonString((PCHAR) raw, &tokens[i], JSMN_STRING, (PCHAR) CANARY_BIT_RATE_ENV_VAR)) {
+            jsonUint64(raw, tokens[++i], &bitRate);
+        } else if (compareJsonString((PCHAR) raw, &tokens[i], JSMN_STRING, (PCHAR) CANARY_FRAME_RATE_ENV_VAR)) {
+            jsonUint64(raw, tokens[++i], &frameRate);
+        } else if (compareJsonString((PCHAR) raw, &tokens[i], JSMN_STRING, (PCHAR) ACCESS_KEY_ENV_VAR)) {
```

Comment:
Do we want access key, secret key and session token as jsons? 
I think we should not give a json capability to avoid any accidental git pushes by any customer of the json file.

Reply:
That's a good point. I'll change that.

Reply:
Updated.